Feature Branch for v2.4.6

attackant opened the pull request:
## Version 2.4.6

### Bugs Fixed

- #1057 - Resolved type error on Automation admin screen
- #1056 - Resolved error fetching list of sections 
- #1055 - Moved async functions outside useEffect to avoid synchronous errors
- #1051 - Updated npm dependencies to match WordPress
- #1050 - Resolved uncaught promise error when saving automation rules
- #1048 - Fix posting bug caused by invalid document name. Thanks @alamedapost for your help!
- #1030 - Fixed render error with "publish-to-apple-news" plugin. Thanks @interactivebytes for reporting!
- #1028 - Fixed inability to select checklist options in Post Apple News Options plugin. Thanks @aeroplanejane for reporting!
- #1018 - Fixed bug deleting automation rules which removed all page contents. Thanks @jtroynousky for reporting!
- #1014 - Resolved invalid document exception while processing articles. Thanks @misaalami for your help!  

### Enhancements 

- #1024 - Upgraded to Node.js 20 and npm 10

### Documentation

- #1042 - Added info for using plugin with News Preview app. Thanks @benlk for the documentation!

attackant committed:
build

attackant committed:
Update version number to 2.4.6

The version number in package.json, apple-news.php, class-apple-news.php, and readme.txt files has been updated from 2.4.5 to 2.4.6.

attackant committed:
Update Node, npm versions and dependencies

Updated Node.js and npm to versions 18 and 9 respectively. Updated versions of many npm packages in both devDependencies and dependencies in package.json. Various version updates were also reflected in package-lock.json.

attackant committed:
Update Node.js version in GitHub workflow

The version of Node.js used in the GitHub workflow has been updated to version 18. This ensures consistency with the recently updated versions in other areas of the project.

attackant committed:
Update Node.js version in GitHub workflow

The version of Node.js used in the GitHub workflow has been increased to version 18. This change aligns the version used in the workflow with those used in other parts of the application.

attackant committed:
Update npm version in package.json

The npm version required for the application has been updated from 9 to 10. This version change ensures consistency and compatibility with other parts of the application infrastructure.

attackant committed:
Upgrade node version in workflows and package.json

The node version used in the GitHub workflows and specified in package.json has been upgraded from 18 to 20. The npm version range has also been slightly adjusted in package.json. The modifications intend to keep the system updated and address potential compatibility issues.

attackant committed:
Update multiple "@wordpress" dependencies

The versions of various "@wordpress" dependencies in both "package.json" and "package-lock.json" files are downgraded to match WP 6.4.2. This includes a switch to previous versions for packages such as "@wordpress/dependency-extraction-webpack-plugin", "@wordpress/a11y", "@wordpress/annotations", among others. Adjustments aim to improve compatibility and address potential issues with the specific project's environment or other dependencies.

attackant committed:
Update frontend.jsx to use new React root API

The 'frontend.jsx' file under 'admin-settings' in the assets/js directory now uses the new React root API for rendering. The root API creates a root from a container and allows rendering operations to be scheduled over time. Instead of rendering directly in the component, a constant container is declared and then a root is created from this container for the rendering process.

attackant committed:
Refactor functional components to use function keyword

This change transforms functional components across several files to use the 'function' keyword instead of arrow functions. It adheres to the project coding style guide and aids in debugging by providing more descriptive function names in stack traces. The code refactoring was applied consistently for all functional components in affected files. No functionality changes were made.

attackant committed:
Update Node.js version in package.json and workflow files

This commit increases the maximum acceptable Node.js version in package.json and updates the Node.js versions in the node-tests.yml and built-release.yml workflow files to version 21. This change helps maintain software alignment with updated technologies and enhances compatibility.

attackant committed:
Major cleanup of npm dependencies

Numerous package dependencies have been updated in package-lock.json. Significant changes largely involve updates to allow for newer versions of packages, some removals due to redundancy or changes in codebase, and a few version-specific updates. This will help maintain alignment with contemporary packages.

attackant committed:
Update term ID type handling in fetchTerm function

The function fetchTerm in use-term-cache service now accepts term ID as a string. It then converts this string to a number and verifies its validity before proceeding. This update improves error handling by ensuring the term ID received is a valid number.

attackant committed:
Merge pull request #1054 from alleyinteractive/issue-1053

Update term ID type handling in fetchTerm function

attackant committed:
build

attackant committed:
Remove async-await in term-selector API fetch

Refactored the API fetch in the term-selector component to remove the asynchronous await functionality. Instead, the search results are directly set, which should improve speed and performance. Please note this might change the sequence of execution in this block of code.

attackant committed:
Merge pull request #1052 from alleyinteractive/issue-1051

Update Node, npm versions and dependencies

attackant committed:
Merge branch 'release/v2.4.6' into issue-1055

attackant committed:
Refactor API fetch in term-selector and use-taxonomies components

Eliminated asynchronous await function in the term-selector component and reorganized code in the use-taxonomies component for clarity and better performance. The changes involve directly setting the search results in term-selector and restructuring some segments in use-taxonomies, although the sequence of execution may have been affected.

attackant committed:
Merge pull request #1057 from alleyinteractive/issue-1055

Issue 1055

attackant committed:
Prevent error by adding check before displaying notices

Updated the useEffect in sidebar.jsx to ensure 'notices' is an array and has content before attempting to display notifications. This aims to prevent sporadic errors related to 'notices' being undefined or empty, with helpful logging added for potential future debugging needs.

attackant committed:
Merge pull request #1058 from alleyinteractive/issue-1030

Prevent error by adding check before displaying notices

renatonascalves committed:
Make the default theme notice dismissible

attackant committed:
Merge branch 'release/v2.4.6' into feature/dismissable-notice

renatonascalves committed:
Merge pull request #1059 from alleyinteractive/feature/dismissable-notice

Make the default theme notice dismissible

attackant committed:
build

attackant committed:
Merge branch 'release/v2.4.6' into issue-1048

attackant committed:
Replace qualifiers with imports

attackant committed:
Merge remote-tracking branch 'origin/issue-1048' into issue-1048

attackant committed:
build

attackant committed:
Refactor `class-push.php` code to use imported libraries/classes

The changes in this commit include refactoring the `class-push.php` file to directly use imported libraries/classes. It replaces the fully qualified class names with their respective aliases through the import statements at the top of the file. This simplifies the code and logical flow, and improves readability for developers. Redundant variables were also removed as part of the modifications.

attackant committed:
Reverting some changes due to unexpected phpunit failures

attackant committed:
Update condition check and get method in settings to fix bundle URL bug

The code is updated to check if the filename is `null` instead of an empty string. This is a more accurate way to verify if filename is set before bundling source. Additionally, the way settings are referred to has been changed. Now the magic method `__get` is used instead of the `get` method, to ensure proper retrieval of the property regardless the visibility of the property.

attackant committed:
Merge pull request #1060 from alleyinteractive/issue-1048

Fixes 1048

attackant commented on admin/apple-actions/class-action.php:
```diff
@@ -48,6 +48,6 @@ abstract public function perform();
 	 */
 	protected function get_setting( $name ) {
-		return $this->settings->get( $name );
+		return $this->settings->__get( $name );
```
`get` is deprecated in favor of `__get`

attackant commented on admin/class-admin-apple-themes.php:
```diff
@@ -574,7 +574,9 @@ public function theme_nag() {
+			),
+			get_current_user_id(),
+			true
 		);
```
Makes notice dismissable

attackant committed:
Update changelog in Apple News plugin readme

Comprehensive and detailed updates have been made to the readme file's changelog for the Apple News plugin. Several bugs were fixed, Node.js and npm versions were upgraded, and some valuable info regarding plugin usage with News Preview app was added.

mslinnea commented on package.json:
```diff
@@ -1,11 +1,11 @@
-    "node": "16",
-    "npm": "8"
+    "node": ">=18.0.0 <22.0.0",
+    "npm": ">=9.0.0 <11.0.0"
```
Is a range necessary?  We had a problem on another project caused by a range, as some packages may load different versions.

attackant replied:
Thanks Linnea, I just made a change to tighten down the ranges, but I think keeping them is a good idea in order to include point releases in within the one version we are on now, in this case 20. Please give the changes a re-review with you thoughts. Thanks!

attackant replied:
Removed the range

mslinnea commented on readme.txt:
```diff
@@ -46,6 +46,21 @@ Please visit our [wiki](https://github.com/alleyinteractive/apple-news/wiki) for
+* Bugfix: #1028 - Fixed inability to select checklist options in Post Apple News Options plugin.
+* Bugfix: #1018 - Fixed bug deleting automation rules which removed all page contents.
+* Bugfix: #1014 - Resolved invalid document exception while processing articles.
+* Enhancement: Upgraded to Node.js 20 and npm 10.
```
Could the engines be set to these specifically?

attackant replied:
See my last commit, I restricted to a major version but still allowing point releases in the range.

attackant replied:
Done

mslinnea reviewed (approved):
Approving but consider that specific node and npm versions may prevent issues.

attackant committed:
Update node, npm versions and downgrade dependencies

This update includes modifying the minimum required versions for node and npm in the Apple News plugin. Additionally, various dependencies associated with WordPress have been downgraded to match WP 6.4. Node version ranges have also been changed in the Github workflows.

attackant committed:
Update node, npm version specifications in Apple News plugin

The node and npm version specifications in package.json and package-lock.json files of the Apple News plugin are modified. Instead of specifying a fixed range, a more flexible "^20.x.x" for node and "^10.x.x" for npm are now used, indicating any minor version updates within the major versions 20 and 10 respectively are acceptable.

mslinnea reviewed (approved):
Looks good! 🐴
